### PR TITLE
Image filter

### PR DESCRIPTION
**Pull request by acastellane**

### CODE REVIEW

File: actions/hls_helloworld_1024/include/action_changecase.h [hunk 1]
```diff
@@ -43,11 +43,11 @@ extern "C" {
...
 // 2. They will be extracted by hardware/setup/patch_version.sh
 // 3. And put into snap_global_vars.v
-// 4. Used by hardware/hls/action_wrapper.v
+// 4. Used by hardware/hdl/hls/action_wrapper.v
 #define ACTION_TYPE               0x10143009
 #define RELEASE_LEVEL             0x00000010
-// For oc_maint, Action descriptions are decoded with the help of software/tools/snap_actions.h
-// Please modify ActionTypes.md file so oc_maint can recognize this action.
+// For snap_maint, Action descriptions are decoded with the help of software/tools/snap_actions.h
+// Please modify this file so snap_maint can recognize this action.
```
Comment (bmesnet) [inline actions/hls_helloworld_1024/include/action_changecase.h:50]: hmm strange to speak about `snap_maint `in oc-accel!

File: actions/hls_image_filter/hw/action_pixel_filter.cpp [hunk 2]
```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright 2017 International Business Machines
+ *
```
Comment (bmesnet) [inline actions/hls_image_filter/hw/action_pixel_filter.cpp:3]: Copyright 2017 => Copyright 2020?

File: actions/hls_image_filter/hw/action_pixel_filter.H [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#ifndef __ACTION_PIXEL_FILTER_H__
+#define __ACTION_PIXEL_FILTER_H__
+
+/*
+ * Copyright 2017 International Business Machines
+ *
```
Comment (bmesnet) [inline actions/hls_image_filter/hw/action_pixel_filter.H:6]: Copyright 2017 => Copyright 2020?

File: actions/hls_image_filter/hw/Makefile [hunk 4]
```diff
@@ -0,0 +1,46 @@
+#
+# Copyright 2017 International Business Machines
+#
```
Comment (bmesnet) [inline actions/hls_image_filter/hw/Makefile:3]: Copyright 2017 => Copyright 2020?

File: actions/hls_image_filter/include/action_search.h [hunk 5]
```diff
@@ -0,0 +1,64 @@
+#ifndef __ACTION_SEARCH_H__
```
Comment (bmesnet) [inline actions/hls_image_filter/include/action_search.h:1]: Is this file used for image filtering?

File: actions/hls_image_filter/sw/bmp.c [hunk 6]
```diff
@@ -0,0 +1,464 @@
+/*
```
Comment (bmesnet) [inline actions/hls_image_filter/sw/bmp.c:1]: No copyright header?

File: actions/hls_image_filter/sw/bmp.h [hunk 7]
```diff
@@ -0,0 +1,65 @@
+#ifndef _BMP_H_  // prevent recursive inclusion
```
Comment (bmesnet) [inline actions/hls_image_filter/sw/bmp.h:1]: no copyright header?

File: actions/hls_image_filter/sw/params.h [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+#include <malloc.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/time.h>
+#include <assert.h>
+#include <osnap_tools.h>
+#include <libosnap.h>
+#include <action_pixel_filtering.h>
+#include "../include/action_pixel_filtering.h"
+#include <osnap_hls_if.h>
```
Comment (bmesnet) [inline actions/hls_image_filter/sw/params.h:35]: do we really need the relative path to` ../include/action_pixel_filtering.h`?

**Review by bmesnet — COMMENTED** [review]
see comments in line

File: actions/hls_image_filter/include/action_search.h (hunk 5 above)
Comment (acastellane) [reply actions/hls_image_filter/include/action_search.h]: removed

File: actions/hls_image_filter/README.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
+# HLS_IMAGE_FILTER EXAMPLE
+
+* Provides a simple base allowing to discover SNAP
```
Comment (bmesnet) [inline actions/hls_image_filter/README.md:3]: he he he ...to discover SNAP?

File: actions/hls_image_filter/hw/action_pixel_filter.H [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+typedef char word_t[BPERDW_1024];
+typedef snapu64_t address_t;
+
+#define PATTERN_SIZE BPERDW_1024
+#define TEXT_SIZE    4096 * MAX_NB_OF_BYTES_READ // used for streaming only
+
+typedef char word_t[BPERDW_1024];
+//---------------------------------------------------------------------
+// This is generic. Just adapt names for a new action
+// CONTROL is defined and handled by SNAP
```
Comment (bmesnet) [inline actions/hls_image_filter/hw/action_pixel_filter.H:41]: CONTROL is defined and handled by SNAP => OC_ACCEL

**Review by bmesnet — DISMISSED** [review]

File: actions/hls_image_filter/README.md (hunk 9 above)
Comment (acastellane) [reply actions/hls_image_filter/README.md]: ok

File: actions/hls_image_filter/hw/action_pixel_filter.H (hunk 10 above)
Comment (acastellane) [reply actions/hls_image_filter/hw/action_pixel_filter.H]: ok

**Review by bmesnet — APPROVED** [review]

**Pull request merged**